pkg/cmd/taskrun/delete.go

```diff
@@ -143,16 +167,25 @@ func deleteTaskRuns(s *cli.Stream, p cli.Params, trNames []string, opts *options
 	return d.Errors()
 }
 
-func taskRunLister(p cli.Params, keep int, cs *cli.Clients) func(string) ([]string, error) {
+func taskRunLister(p cli.Params, keep int, kind string, cs *cli.Clients) func(string) ([]string, error) {
 	return func(taskName string) ([]string, error) {
+		label := "task"
+		if kind == "ClusterTask" {
+			label = "clusterTask"
+		}
+
 		lOpts := metav1.ListOptions{
-			LabelSelector: fmt.Sprintf("tekton.dev/task=%s", taskName),
+			LabelSelector: fmt.Sprintf("tekton.dev/%s=%s", label, taskName),
 		}
 		trs, err := trlist.TaskRuns(cs, lOpts, p.Namespace())
 		if err != nil {
 			return nil, err
 		}
-		trs.Items = taskpkg.FilterByRef(trs.Items, string(v1beta1.NamespacedTaskKind))
+		if kind == "ClusterTask" {
```

Comment:
we need the filter only if the kind is Task

Reply:
We need this for `ClusterTask` as well right since we need to filter the `TaskRuns` by `ClusterTask` also?

Reply:
No, we don't need it because taskruns associated with cluster tasks can be filtered by tekton.dev/clustertask and this will not be there for taskrun started by the task

Reply:
ohh..okay...got it :+1: